<launch>

  <arg name="real_robot" default="false"/>

  <!-- ROS-Pybullet interface -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface" type="ros_pybullet_interface_node.py" output="screen">
    <rosparam param="config" file="$(find rpbi_examples)/configs/basic_example_kuka_lbr/config.yaml"/>
  </node>

  <!-- Static transforms -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="floor_tf" args="0 0 0 0 0 0 rpbi/world rpbi/floor"/>

  <group if="$(arg real_robot)">

    <!-- Remapper (for real robot) -->
    <node pkg="custom_ros_tools" type="remap_joint_state_to_joint_state" name="remapper">
      <remap from="joint_states/from" to="rpbi/kuka_lbr/joint_states"/>
      <remap from="joint_states/to" to="lbr/joint_states/command"/>    
    </node>
    
    <!-- <node pkg="moveit_python_examples" name="joint_state_command" type="stream_joint_state_node.py" ns="lbr" output="screen"/> -->

  </group>
    

  <!-- Example node that generates target joint positions -->
  <node pkg="rpbi_examples" name="example" type="basic_robot_example_node.py" args="kuka_lbr" output="screen">
    <param name="real_robot" type="bool" value="$(arg real_robot)"/>
    <param name="real_robot_joint_states_topic" type="str" value="lbr/joint_states"/>
  </node>

</launch>
